src/components/Storage/api/usePath.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useHistory, useParams } from 'react-router-dom';
+
+import { storagePath } from '../common/constants';
+import { StorageRouterParams } from '../types';
+import { useBucket } from './useBucket';
+
+export function usePath(): [string, (bucket: string) => void] {
+  const { path } = useParams<StorageRouterParams>();
+  const [bucket] = useBucket();
+  const history = useHistory();
+
+  function setPath(path: string) {
+    while (path.includes('//')) {
```

Comment:
This should just use regex for global replace `const MULTI_SLASH_GLOBAL = new RegExp('//+', 'g')`